windows/win_scheduled_task.ps1

```diff
@@ -33,40 +34,161 @@ else
 {
     Fail-Json $result "missing required argument: name"
 }
+if ($params.state)
+{
+    $state = $params.state
+}
+else
+{
+    Fail-Json $result "missing required argument: state"
+}
 if ($params.enabled)
 {
   $enabled = $params.enabled | ConvertTo-Bool
 }
 else
 {
-  $enabled = $true
+  $enabled = $true #default
 }
-$target_state = @{$true = "Enabled"; $false="Disabled"}[$enabled]
-
-try
+if ($params.description)
 {
-  $tasks = Get-ScheduledTask -TaskPath $name
-  $tasks_needing_changing = $tasks |? { $_.State -ne $target_state }
-  if (-not($tasks_needing_changing -eq $null))
-  {
-    if ($enabled)
-    {
-      $tasks_needing_changing | Enable-ScheduledTask
+  $description = $params.description
+}
+else
+{
+  $description = " "  #default
+}
+if ($params.execute)
+{
+  $execute = $params.execute
+}
+elseif ($state -eq "present")
+{
+  Fail-Json $result "missing required argument: execute"
+}
+if( $state -ne "present" -and $state -ne "absent") {
+    Fail-Json $result "state must be present or absent"
+}
+if ($params.path)
+{
+  $path = "\{0}\" -f $params.path
+}
+else
+{
+  $path = "\"  #default
+}
+if ($params.frequency)
+{
+  $frequency = $params.frequency
+}
+elseif($state -eq "present")
+{
+  Fail-Json $result "missing required argument: frequency"
+}
+if ($params.time)
+{
+  $time = $params.time
+}
+elseif($state -eq "present")
+{
+  Fail-Json $result "missing required argument: time"
+}
+if ($params.daysOfWeek)
+{
+  $daysOfWeek = $params.daysOfWeek
+}
+elseif ($frequency -eq "weekly")
+{
+  Fail-Json $result "missing required argument: daysOfWeek"
+}
+
+try {
+    $task = Get-ScheduledTask -TaskPath "$path" | Where-Object {$_.TaskName -eq "$name"}
+
+    # Correlate task state to enable variable, used to calculate if state needs to be changed
+    $taskState = $task.State
+    if ($taskState -eq "Ready"){
+        $taskState = $true
+    }
+    elseif($taskState -eq "Disabled"){
+        $taskState = $false
     }
     else
     {
-      $tasks_needing_changing | Disable-ScheduledTask
+        $taskState = $null
+    }
+
+    $measure = $task | measure
+    if ($measure.count -eq 1 ) {
+        $exists = $true
+    }
+    elseif ($measure.count -eq 0 -and $state -eq "absent" ){
+        Set-Attr $result "msg" "Task does not exist"
+        Exit-Json $result
+    }
+    elseif ($measure.count -eq 0){
+        $exists = $false
+    }
+    else {
+        # This should never occur
+        Fail-Json $result "$measure.count scheduled tasks found"
```

Comment:
I'm not sure how this condition would ever occur (the UI and powershell won't let you create 2 tasks with the same name). That being said I updated the line.